cli/command/environment_test.go

```diff
@@ -0,0 +1,381 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/quintilesims/layer0/api/job"
+	"github.com/quintilesims/layer0/cli/printer"
+	"github.com/quintilesims/layer0/cli/resolver/mock_resolver"
+	"github.com/quintilesims/layer0/client/mock_client"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+)
+
+func TestCreateEnvironment_userInputErrors(t *testing.T) {
+	_, _, command, _ := initEnvCommandTest(t)
+
+	testCases := []struct {
+		name    string
+		command func(*cli.Context) error
+		args    []string
+	}{
+		{
+			name:    "create",
+			command: command.create,
+		},
+		{
+			name:    "update",
+			command: command.update,
+		},
+		{
+			name:    "setMinCount",
+			command: command.update,
+			args:    []string{"NAME", "1w"},
+		},
+		{
+			name:    "read",
+			command: command.read,
+		},
+		{
+			name:    "delete",
+			command: command.delete,
+		},
+		{
+			name:    "link",
+			command: command.link,
+		},
+		{
+			name:    "unlink",
+			command: command.unlink,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			c := GetCLIContext(t, tc.args, nil)
+
+			if err := tc.command(c); err == nil {
+				t.Fatalf("%s: error was nil!", tc.name)
+			}
+		})
+	}
+}
+
+func TestCreateEnvironment(t *testing.T) {
```

Comment:
Seems like we're forcing a table test here. With only 2 different inputs, and the significant difference of behavior between them, seems like these should just be 2 tests. 

edit: this seems to apply to add wait vs. non-wait tests

Reply:
Yes definitely agree it is forcing a table test here. I left it in to argue a good reason for it to exist as a table test though. 

All the commands that return a job have a wait/non-wait test which is essentially just the same method with 2 additional method call expectations. If someone was trying to figure out what was different between the two tests that almost look identical, they would have to scroll between the two methods and figure out how they differ.

Whereas passing in the `no-wait` flag to the test, it is immediately clear how the code differs when the `no-wait` flag is and isn't used.

Is that a convincing enough argument. I convinced myself into this.

Reply:
i think this is a good candidate for a generalized helper for all wait vs. non-wait tests

Reply:
@zpatrick I've added a generalized helper here. Thoughts? https://github.com/quintilesims/layer0/pull/384/files#diff-49f6506166fd008153f625177a0eaa5bR85

Reply:
this is nice! some small things:
* a nitpick, but perhaps `testWaitHelper` is more appropriate? `testNoWaitHelper` seems to imply it's only for no-wait=true
* the inner function's middle param should be `flags Flags`. The body of this test shouldn't re-create that object, but just append to it:
```
testWaitHelper(t, func(t *testing.T, flags Flags, wait bool){
    ...
    flags["size"] = req.InstanceSize
    flags["min-count"] = req.MinClusterCount
    ...
}
```

Reply:
✅